core/src/main/scala/ai/lum/odinson/utils/exceptions/OdinsonException.scala

```diff
@@ -5,14 +5,13 @@ package ai.lum.odinson.utils.exceptions
   * @param message
   * @param cause
   */
-class OdinsonException(message: String = null, cause: Throwable = null)
-    extends RuntimeException(OdinsonException.defaultMessage(message, cause), cause) {}
+class OdinsonException(message: String, cause: Throwable) extends RuntimeException(message) {
+  if (cause != null) { initCause(cause) }
+  def this(message: String) = this(message, null)
+}
 
 object OdinsonException {
-
-  def defaultMessage(message: String, cause: Throwable) =
-    if (message != null) message
-    else if (cause != null) cause.toString()
-    else null
-
-}
+  def apply(message: String, cause: Throwable): OdinsonException = {
```

Comment:
It's not a big deal, but I think this might encourage people to use null as a second argument and it seems like we're trying to avoid those.  Perhaps there could be two apply methods:
```
apply(message: String) = new OdinsonException(message, None)
apply(message: String, cause: Throwable) = new OdinsonException(message, Some(throwable))
```

Reply:
Sure.